pkg/oc/cli/admin/mustgather/mustgather.go

```diff
@@ -0,0 +1,367 @@
+		Use:     "must-gather",
+		Short:   "Launch a new instance of a pod for gathering debug information",
+		Long:    mustGatherLong,
+		Example: mustGatherExample,
+		Hidden:  true,
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, cmd, args))
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+
+	cmd.Flags().StringVar(&o.NodeName, "node-name", "", "Set a specific node to use - by default a random master will be used")
+	cmd.Flags().StringVar(&o.Image, "image", "", "Set a specific image to use, by default the OpenShift's must-gather image will be used.")
+	cmd.Flags().StringVar(&o.DestDir, "dest-dir", "", "Set a specific directory on the local machine to write gathered data to.")
+	cmd.Flags().BoolVar(&o.Keep, "keep", false, "Do not delete temporary resources when command completes.")
+	_ = cmd.Flags().MarkHidden("keep")
+
+	return cmd
+}
+
+func (o *Options) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	if i := cmd.ArgsLenAtDash(); i != -1 && i < len(args) {
+		o.Command = args[i:]
+	} else {
+		o.Command = args
+	}
+
+	var err error
+
+	if o.ClientConfig, err = f.ToRESTConfig(); err != nil {
+		return err
+	}
+
+	if o.KubeClient, err = kubernetes.NewForConfig(o.ClientConfig); err != nil {
+		return err
+	}
+
+	if len(o.DestDir) == 0 {
+		o.DestDir = fmt.Sprintf("must-gather.local.%06d", rand.Int63())
+	}
+
+	if len(o.Image) == 0 {
+		// TODO lookup cluster specific default
+		o.Image = "quay.io/openshift/origin-must-gather:v4.0"
+	}
+
+	return nil
+}
+
+type Options struct {
+	genericclioptions.IOStreams
+
+	ClientConfig *rest.Config
+	KubeClient   kubernetes.Interface
+
+	NodeName string
+	DestDir  string
+	Image    string
+	Command  []string
+	Keep     bool
+}
+
+// Run creates and runs a must-gather pod.d
+func (o *Options) Run() error {
+	if len(o.ClientConfig.BearerToken) != 0 {
+		return fmt.Errorf("cannot run must-gather with a token based kubeconfig")
+	}
+	if len(o.Image) == 0 {
+		return fmt.Errorf("missing an image")
+	}
+
+	// TODO This command will:
+	// [x] Create a temporary namespace
+	// [x] Create a secret an upload the local kubeconfig content into it
+	// [ ] Look up the image to use for gathering information
+	// [x] Create a pod that runs the 'gather' command in that image
+	// [x] Wait for completion of that command
+	// [xs] Copy the gathered data locally
+
+	var err error
+
+	// create namespace
+	ns, err := o.KubeClient.CoreV1().Namespaces().Create(&corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "openshift-must-gather-",
+			Labels: map[string]string{
+				"openshift.io/run-level": "0",
+				"app": "must-gather",
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	if !o.Keep {
+		defer func() {
+			err = o.KubeClient.CoreV1().Namespaces().Delete(ns.Name, nil)
+		}()
+	}
+
+	clusterRoleBinding, err := o.KubeClient.RbacV1().ClusterRoleBindings().Create(o.newClusterRoleBinding(ns.Name))
+	if err != nil {
+		return err
+	}
+	if !o.Keep {
```

Comment:
printf indicating what clusterrolebinding.rbac.authorization.k8s.io we created

Reply:
Done